networked_controller.gd

```diff
@@ -53,20 +58,23 @@ func controller_process(delta: float, db: DataBuffer):
 	# Process the controller.
 
 	# Take the inputs
+	var is_jumping = db.read_bool()
 	var input_direction := Vector2()
 
 	var has_input = db.read_bool()
 	if has_input:
 		input_direction = db.read_normalized_vector2(DataBuffer.COMPRESSION_LEVEL_3)
 
 	# Process the character
-	get_parent().step_body(delta, Vector3(input_direction.x, 0.0, input_direction.y))
+	get_parent().step_body(delta, Vector3(input_direction.x, 0.0, input_direction.y), is_jumping)
 
 
 func count_input_size(inputs: DataBuffer) -> int:
 	# Count the input buffer size.
 	var size: int = 0
 	size += inputs.get_bool_size()
+	size += inputs.get_bool_size()
+	inputs.skip_bool()
```

Comment:
I would invert this, so to have:
```
size += inputs.get_bool_size()
inputs.skip_bool()

size += inputs.get_bool_size()
if inputs.read_bool():
```

It should be cleaner.

Reply:
Agree!